xunit-autolabeler-v2/ast_parser/core/analyze_test.py

```diff
@@ -0,0 +1,134 @@
+# Copyright 2020 Google LLC. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import os
+import unittest
+
+from ast_parser.core import analyze
+
+import mock
+
+import pytest
+
+
+TEST_DIR = os.path.join(
+    os.path.dirname(__file__),
+    'test_data/parser'
+)
+
+
+class AnalyzeBasicTests(unittest.TestCase):
+    @pytest.fixture(autouse=True)
+    def _get_analyze_result(self):
+        basic_test_dir = os.path.join(TEST_DIR, 'edge_cases')
+        analyze_result = analyze.analyze_json(
+            os.path.join(basic_test_dir, 'polyglot_snippet_data.json'),
+            basic_test_dir
+        )
+
+        self.grep_tags = analyze_result[0]
+        self.source_tags = analyze_result[1]
+        self.ignored_tags = analyze_result[2]
+        self.source_methods = analyze_result[3]
+
+    def test_normalizes_source_paths(self):
+        assert self.source_methods != []
```

Comment:
assert len(self.source_methods) != 0
to assert a list is not empty.

Reply:
[PEP 8](https://stackoverflow.com/a/53752) asserts this is an anti-pattern. (Happy to do it, but wanted to double-check.)

Reply:
Mmm I guess it depends on what you are trying to assert here, self.source_methods truthiness or whether it is empty?
My worry with the current implementation is that any unexpected non-empty-list value for self.source_methods will pass this assertion.

Reply:
This is mostly a "precondition" for other parts of the test. (The core of the test is the last two assertions.) It's checking for emptiness, since this test's primary goal is to validate `source_methods[<something>].source_path`.)

FWIW I'm happy to remove this line, if you don't mind tests throwing exceptions instead of "over-asserting" + gracefully failing. (The passing condition won't change.)